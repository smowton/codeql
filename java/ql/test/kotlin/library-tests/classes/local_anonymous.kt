package LocalAnonymous

class Class1 {
    fun fn1(): Any {
        return object {
            fun fn() {}
        }
    }

    fun fn2() {
        fun fnLocal() {}
        fnLocal()
    }

    fun fn3() {
        val lambda1 = { a: Int, b: Int -> a + b }
        val lambda2 = fun(a: Int, b: Int) = a + b
    }

    fun fn4() {
        val fnRef = Class1::fn3
    }

    fun fn5() {
        class LocalClass {}
        LocalClass()
    }
}
