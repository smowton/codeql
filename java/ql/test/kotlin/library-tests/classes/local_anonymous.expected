anonymousObjects
| local_anonymous.kt:5:16:7:9 | new (...) | local_anonymous.kt:5:16:7:9 | new Object(...) { ... } | anonymous | local |
localFunctions
| local_anonymous.kt:11:9:11:24 | fnLocal | local_anonymous.kt:11:9:11:24 | new Object(...) { ... } | anonymous | local |
lambdas
| local_anonymous.kt:16:23:16:49 | ...->... | local_anonymous.kt:16:23:16:49 | new Function2<Integer,Integer,Integer>(...) { ... } | anonymous | not local |
| local_anonymous.kt:17:23:17:49 | ...->... | local_anonymous.kt:17:23:17:49 | new Function2<Integer,Integer,Integer>(...) { ... } | anonymous | not local |
memberRefs
| local_anonymous.kt:21:21:21:31 | ...::... | local_anonymous.kt:21:21:21:31 | new Function0<Unit>(...) { ... } | anonymous | not local |
localClasses
| local_anonymous.kt:25:9:25:27 | LocalClass | not anonymous | local |
